### DISCUSSION TITLE
Starting Convo Error:  Loading user data failed: Could not connect.

### DISCUSSION THREAD

**Discussion by vektren** [convos-chat/convos#794] [General]
Hello, 
I have set up Convos on Ubuntu Server that runs on Oracle Cloud. 
I can access the page through Nginx reverse proxy:  https://irc.epicsheep.com.

The page loads with no problem with the error: Loading user data failed: Could not connect. And then it just retries? 

To test, I opened all ports on the firewall I have looked for connection files. 
I am sure I am just missing something small.

**Comment by jhthorsen** [comment]
There could be multiple issues:

* Have you made sure that the appropriate environment variables have been set and that nginx sets the correct headers? Please have a look at https://convos.chat/doc/reverse-proxy for more details
* I *think* some browsers does not want to make a WebSocket connection if you have self-signed certificates. I would suggest using https://letsencrypt.org/ to make sure you have valid certificates.

**Comment by abackstrom** [comment]
This is also pretty hit-or-miss for me. I'm successfully using Convos on a few devices, but I usually get stuck in a loop trying to log a new device in. The backend repeatedly sends `[2025-11-19 21:58:39.30468] [2375446] [error] [fallback] [Convos::Controller::Events] Need to log in first.` while the browser console shows this error over and over:

<img width="2560" height="1440" alt="Screenshot from 2025-11-19 21-55-57" src="https://github.com/user-attachments/assets/c556041f-f564-438f-b2ff-6afec1ac9f1c" />

Resetting site data and manually deleting everything from storage does not help. Eventually I'll get the login form, seemingly without any input on my end.

I am running Convos behind a Caddy server with a Let's Encrypt cert.
